odoo/tools/cache.py

```diff
@@ -191,3 +227,35 @@ def get_cache_key_counter(bound_method: Callable, *args, **kwargs):
     cache, key0, counter = ormcache_instance.lru(model)
     key = key0 + ormcache_instance.key(model, *args, **kwargs)
     return cache, key, counter
+
+
+def get_cache_footprint(obj, * , seen_ids: set[int] | None = None, cache_info: str) -> int:
+    """ A non-thread-safe recursive object size estimator """
+    from odoo.models import BaseModel
+    from odoo.api import Environment
+
+    if seen_ids is None:
+        seen_ids = set()
+    total_size = 0
+    objects = [obj]
+
+    while objects:
+        cur_obj = objects.pop()
+        if id(cur_obj) in seen_ids or callable(cur_obj):
+            continue
+
+        if isinstance(cur_obj, (BaseModel, Environment)):
+            _logger.error(f'{cur_obj} is cached by {cache_info}')
```

Comment:
Logging should not use f-strings.
```suggestion
            _logger.error('%s is cached by %s', cur_obj, cache_info)
```